subsys/jaxrs/src/main/java/org/commonjava/indy/bind/jaxrs/ApiVersioningFilter.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (C) 2011-2017 Red Hat, Inc. (https://github.com/Commonjava/indy)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.bind.jaxrs;
+
+import org.commonjava.indy.stats.IndyVersioning;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+import static org.commonjava.indy.stats.IndyVersioning.HEADER_INDY_API_VERSION;
+import static org.commonjava.indy.util.ApplicationHeader.accept;
+
+@ApplicationScoped
+public class ApiVersioningFilter
+                implements Filter
+{
+
+    private final Logger logger = LoggerFactory.getLogger( getClass() );
+
+    @Inject
+    private IndyVersioning indyVersioning;
+
+    public ApiVersioningFilter()
+    {
+    }
+
+    public ApiVersioningFilter( IndyVersioning versioning )
+    {
+        this.indyVersioning = versioning;
+    }
+
+    @Override
+    public void init( final FilterConfig filterConfig ) throws ServletException
+    {
+    }
+
+    @Override
+    public void doFilter( final ServletRequest request, final ServletResponse response, final FilterChain chain )
+                    throws IOException, ServletException
+    {
+        HttpServletRequest httpServletRequest = (HttpServletRequest) request;
+        String reqApiVersion = httpServletRequest.getHeader( HEADER_INDY_API_VERSION );
+
+        VersioningRequest versioningRequest = new VersioningRequest( httpServletRequest, reqApiVersion );
```

Comment:
This needs to be reworked for the new versioning approach. Instead of using Accept header, we're using Indy-API-Version and request pathInfo to lookup a deprecated path. If that deprecated path exists, we will forward to that using RequestDispatcher and inject a deprecation warning in an outgoing header. See our design docs for more information.

Reply:
Ok